### PR TITLE
Add CI pipeline via Github Actions

### LINKED ISSUE #192: Summon pipeline runs using GitHub actions instead of a private Jenkins server
At current, the Summon pipeline runs in Jenkins - but this makes pipeline runs less visible to community members.

In this card, we'll migrate the non-privileged stages of the Summon pipeline to use GH action flows instead.

### PR DESCRIPTION
**Pull request by doodlesbykumbi**
Github Action runs are public which provides much needed visibility into the build status.

### What does this PR do?
- _What's changed? Why were these changes made?_
- _How should the reviewer approach this PR, especially if manual tests are required?_
- _Are there relevant screenshots you can add to the PR description?_

### What ticket does this PR close?
Resolves #192 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: build [hunk 1]
```diff
@@ -9,4 +9,4 @@ docker pull "$GORELEASER_IMAGE"
 docker run --rm \
            -v "$PWD:/summon" \
            -w /summon \
-           "$GORELEASER_IMAGE" release --rm-dist --skip-validate
+           "$GORELEASER_IMAGE" release --rm-dist $@
```
Comment (diverdane) [inline build:12]: Probably better with quotes around `$@` (should still pass individual args in the expanded array).

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Build Go package (release)
+        run: ./build
+        if:  ${{ startsWith(github.ref, 'refs/tags/v') }}
+      - name: Build Go package (snapshot)
+        run: ./build --snapshot
```
Comment (diverdane) [inline .github/workflows/ci.yml:16]: Is the `--snapshot` flag used in the `build` script?

**Review by diverdane — COMMENTED** [review]
Looks good to me. One quick question on the snapshot flag.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (doodlesbykumbi) [reply .github/workflows/ci.yml:16]: It's passed to goreleaser, which takes it as a flag for non-release builds

File: build (hunk 1 above)
Comment (doodlesbykumbi) [reply build]: Done

File: .github/workflows/ci.yml (hunk 2 above)
Comment (diverdane) [reply .github/workflows/ci.yml:16]: Ah, thanks!

**Review by diverdane — APPROVED** [review]
LGTM!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+name: CI
+
+on: [push]
```
Comment (izgeri) [inline .github/workflows/ci.yml:3]: @doodlesbykumbi this won't give us daily runs - can you please add a cronjob for this like we have here: 
https://github.com/cyberark/conjur-quickstart/blob/76dbdbf463c3f4405c4c87dc2607eb89b458c25d/.github/workflows/quick_start_flow_test.yml#L5

**Pull request merged**